### PR TITLE
#18177 fix for the jsp when the title is not being set, used the iden…

### PR DESCRIPTION
**Pull request by jdotcms**
Some content types are having issues when the title is mark as not required and the title is not set in consequence.

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/vanityurl/model/DefaultVanityUrl.java [hunk 1]
```diff
@@ -18,10 +18,6 @@
      */
     private static final long serialVersionUID = 1L;
 
-    @Override
-    public String getTitle() {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/vanityurl/model/DefaultVanityUrl.java:22]: why did you remove this method?

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java [hunk 2]
```diff
@@ -222,15 +222,21 @@ protected long contentletIdentifierCount() throws DotDataException {
 	@Override
 	public com.dotmarketing.portlets.contentlet.business.Contentlet convertContentletToFatContentlet(Contentlet cont,
 			com.dotmarketing.portlets.contentlet.business.Contentlet fatty) throws DotDataException {
+
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java:225]: A test should be implemented

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java [hunk 3]
```diff
@@ -300,12 +303,93 @@ public String getTitle(){
 		}
 	}
 
+	@CloseDBIfOpened
+	private String buildName()
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java:307]: Add a javadoc

**Review by nollymar — CHANGES_REQUESTED** [review]
It is important to add testing to these changes, especially if a legacy test was removed

File: dotCMS/src/main/java/com/dotcms/vanityurl/model/DefaultVanityUrl.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/vanityurl/model/DefaultVanityUrl.java:22]: we redesign the logic; not the getTitle is final. No body can override it. This is to avoid weird behaviour such as the contentlets modified here, that the title generating didn't follow the dotCMS logic

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java [hunk 4]
```diff
@@ -2,30 +2,37 @@
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java:7]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert.assertNull'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5248304)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java:7]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert.assertNull'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5248304)
